client/src/main.rs

```diff
@@ -754,6 +768,12 @@ pub async fn main() -> Result<()> {
 
 	let reason = shutdown.completed_shutdown().await;
 
+	let restart = restart.lock().await;
+	if *restart {
+		info!("Restarting Light Client...");
+		utils::restart();
+	}
```

Comment:
is that `restart` variable used elsewhere, 
if not, can we use something like this?

```suggestion
if *restart.lock().await {
    info!("Restarting Light Client...");
    utils::restart();
}
```